planning/obstacle_cruise_planner/src/planner_interface.cpp

```diff
@@ -143,9 +143,13 @@ Trajectory PlannerInterface::generateStopTrajectory(
   const double closest_obstacle_dist = motion_utils::calcSignedArcLength(
     planner_data.traj.points, 0, closest_stop_obstacle->collision_points.front().point);
 
-  const auto negative_dist_to_ego = motion_utils::calcSignedArcLength(
-    planner_data.traj.points, planner_data.current_pose, 0, ego_nearest_param_.dist_threshold,
+  const auto ego_segment_idx = motion_utils::findNearestSegmentIndex(
+    planner_data.traj.points, planner_data.current_pose, ego_nearest_param_.dist_threshold,
     ego_nearest_param_.yaw_threshold);
+
+  const auto negative_dist_to_ego = motion_utils::calcSignedArcLength(
+    planner_data.traj.points, planner_data.current_pose.position, *ego_segment_idx, 0);
+
   if (!negative_dist_to_ego) {
```

Comment:
This check `if (!negative_dist_to_ego) {` is not required anymore.

Reply:
can i delete this check and lines below?
```
    const auto markers =
      motion_utils::createDeletedStopVirtualWallMarker(planner_data.current_time, 0);
    tier4_autoware_utils::appendMarkerArray(markers, &debug_data.stop_wall_marker);
```